src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcher.cs

```diff
@@ -260,8 +262,8 @@ internal async Task<IEnumerable<ILanguageWorkerChannel>> GetInitializedWorkerCha
                     }
                 }
             }
-            IEnumerable<ILanguageWorkerChannel> workerChannels = webhostChannels == null ? _jobHostLanguageWorkerChannelManager.GetChannels() : webhostChannels.Union(_jobHostLanguageWorkerChannelManager.GetChannels());
-            IEnumerable<ILanguageWorkerChannel> initializedWorkers = workerChannels.Where(ch => ch.State == LanguageWorkerChannelState.Initialized);
+            // TODO: optimize so that if there are multiple language worker channels, prioritize initialized.
+            IEnumerable<ILanguageWorkerChannel> initializedWorkers = webhostChannels == null ? _jobHostLanguageWorkerChannelManager.GetChannels() : webhostChannels.Union(_jobHostLanguageWorkerChannelManager.GetChannels());
```

Comment:
this call is just getting all the channels. You still need to filter the channels that initialized before sending invocation requests.